src/test/java/com/cegeka/academy/repository/InterestTest.java

```diff
@@ -61,4 +67,58 @@ public void testFindByDescription(){
         assertThat(interestRepository.findByDescription("description1").get(0)).isEqualTo(interestRepository.findAll().get(0));
     }
 
+    @Test
+    public void testAddUserInterest(){
+        Interest interest1  = new Interest();
+        interest1.setName("interest1");
+        interest1.setDescription("description1");
+
+        User user = new User();
+        user.setEmail("gigi.gogaie@gmail.com");
+        user.setFirstName("Gogaie");
+        user.setLastName("Momaie");
+        user.setPassword("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+        user.setLogin("gigel@purcel.com");
+
+        Set userSet = new HashSet<>();
+        Set interestSet = new HashSet<>();
+
+        userSet.add(user);
+        interestSet.add(interest1);
+
+        user.setUserInterests(interestSet);
+        interest1.setInterestUsers(userSet);
+
+        interestRepository.save(interest1);
+        userRepository.save(user);
+
+        assertThat(interestRepository.findAll().size()).isEqualTo(1);
+        assertThat(interestRepository.findAll().get(0).getName()).isEqualTo("interest1");
+        assertThat(interestRepository.findAll().get(0).getDescription()).isEqualTo("description1");
+        for(int i =0 ;i < userRepository.findAll().size(); i++)
```

Comment:
Nu e neaparata nevoie de `System.out.println`. Ingreuneaza citirea codului si incarca logurile.

Reply:
I used `System.out.println()` because User table had multiple entries before I inserted mine and I had to see the position of the required User in the list. I can remove this code but I thought that it would be useful in the future if some tests fail to see if the cause is testing the wrong user. I don't know how entries persist at DB level when test are run so I  kept the code in case someone runs a test that adds a user, before my test is executed.

Reply:
Am ramas dator cu niste explicatii:

- noi in teste folosim o baza de date in-memory(H2), aceasta este ridicata goala, se ruleaza scripturile liquibase pe ea, se fac operatiile din testul unitar si este oprita. Liquibase insereaza 3 useri default creati de jhipster.

- in testele unitre nu ar trebui sa fim constransi de starea bazei de date(acum avem 3 useri default, poate pe viitor vom adauga ori elimina), astfel ar trebui sa iti cauti userul tau dupa `login` sau alta proprietate care sa garanteze unicitatea entitatii.